.travis.yml

```diff
@@ -1,30 +1,9 @@
-language: dart
-os:
-  - linux
-sudo: required
-addons:
-  apt:
-    # Flutter depends on /usr/lib/x86_64-linux-gnu/libstdc++.so.6 version GLIBCXX_3.4.18
-    sources:
-      - ubuntu-toolchain-r-test # if we don't specify this, the libstdc++6 we get is the wrong version
-    packages:
-      - libstdc++6
-      - fonts-droid-fallback
-install:
-  - sudo apt-get update
-  - sudo apt-get install apt-transport-https
-  - sudo sh -c 'curl https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'
-  - sudo sh -c 'curl https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'
-  - sudo apt-get update
-  - sudo apt-get install dart
-  - export PATH="$PATH:/usr/lib/dart/bin"
-before_script:
-  - git clone https://github.com/flutter/flutter.git -b beta
-  - cd app_feup
-  - ../flutter/bin/flutter doctor
-  - ../flutter/bin/flutter packages get
-script:
-  - dartanalyzer --fatal-warnings --options analysis_options.yaml lib 
-cache:
-  directories:
-    - $HOME/.pub-cache
+services: docker
+
+jobs:
+  include:
+    - stage: lint & test
+      name: "Linter"
+      script: docker build -t app-feup . && docker run -it app-feup flutter analyze --no-pub --preamble .
+    - script: docker build -t app-feup . && docker run -it app-feup flutter test
```

Comment:
Isn't it possible to build the image just once and use the same build in the static analysis and in the testing? I believe the way it is right now, it is building the image twice in parallel. We could build it once in a "build" stage and then have the "lint & test" stage running on that build.

Reply:
before_script does it!